django_mongodb/compiler.py

```diff
@@ -591,42 +603,26 @@ def execute_sql(self, result_type):
                         f"{field.__class__.__name__}."
                     )
             prepared = field.get_db_prep_save(value, connection=self.connection)
-            values.append((field, prepared))
+            if hasattr(value, "as_mql"):
+                prepared = prepared.as_mql(self, self.connection)
+            values[field.column] = prepared
+        try:
+            criteria = self.build_query().mongo_query
+        except EmptyResultSet:
+            return 0
         is_empty = not bool(values)
-        rows = 0 if is_empty else self.update(values)
+        rows = (
+            0
+            if is_empty
+            else self.collection.update_many(criteria, [{"$set": values}]).matched_count
+        )
         for query in self.query.get_related_updates():
             aux_rows = query.get_compiler(self.using).execute_sql(result_type)
             if is_empty and aux_rows:
                 rows = aux_rows
                 is_empty = False
         return rows
 
-    def update(self, values):
-        spec = {}
-        for field, value in values:
-            if field.primary_key:
-                raise DatabaseError("Cannot modify _id.")
-            if isinstance(value, Expression):
-                raise NotSupportedError("QuerySet.update() with expression not supported.")
-            # .update(foo=123) --> {'$set': {'foo': 123}}
-            spec.setdefault("$set", {})[field.column] = value
-        return self.execute_update(spec)
-
-    @wrap_database_errors
```

Comment:
We lost this decorator.

Reply:
Who needs the safe net?! my bad. 😬 deleted by mistake.